Closes #19840 - Enable Site Filtering for Devices in Cable Bulk Import

pheus opened the pull request:
<!--
    Thank you for your interest in contributing to NetBox! Please note that
    our contribution policy requires that a feature request or bug report be
    approved and assigned prior to opening a pull request. This helps avoid
    waste time and effort on a proposed change that we might not be able to
    accept.

    IF YOUR PULL REQUEST DOES NOT REFERENCE AN ISSUE WHICH HAS BEEN ASSIGNED
    TO YOU, IT WILL BE CLOSED AUTOMATICALLY.

    Please specify your assigned issue number on the line below.
-->
### Fixes: #19840 - Add Site information to CSV Import of cables

This PR addresses issue [#19840](https://github.com/netbox-community/netbox/issues/19840) by introducing the ability to specify `side_a_site` and `side_b_site` when importing cables via CSV. This change enables importing cables between devices with duplicate names located in different sites.

#### Changes

- Adds optional `side_a_site` and `side_b_site` fields to the Cable bulk import form.
- Enhances device lookup logic during import to filter by the specified site when resolving device names.
- Refactors and expands test cases:
  - Introduces test data with multiple sites and devices sharing the same name.
  - Verifies correct cable creation using site-scoped device resolution.
- Updates test CSV files to demonstrate usage of the new fields.

#### Example CSV

```csv
side_a_site,side_a_device,side_a_type,side_a_name,side_b_site,side_b_device,side_b_type,side_b_name
DC1,Switch1,dcim.interface,Ethernet1/2,DC2,Switch1,dcim.interface,Ethernet1/2
```

This CSV will correctly match `device1` in *DC1* to `device1` in *DC2* and create a cable between their `Ethernet1/2` interfaces.

pheus committed:
feat(dcim): Add site fields to Cable bulk import form

Introduces `side_a_site` and `side_b_site` fields for the Cable bulk
import form. Limits device choices on both sides to the selected site
for improved input validation and consistency.

pheus committed:
feat(dcim): Enhance test data setup with multiple sites

Refactors tests to create multiple sites and assign devices accordingly.
Updates CSV data to include `side_a_site` and `side_b_site` fields for
scenarios involving multiple sites. This improves test coverage and
alignment with real-world use cases.

jnovinger commented on netbox/dcim/tests/test_views.py:
```diff
@@ -3266,17 +3266,21 @@ class CableTestCase(
+            Device(name='Device 1', site=sites[0], device_type=devicetype, role=role),
+            Device(name='Device 2', site=sites[0], device_type=devicetype, role=role),
+            Device(name='Device 3', site=sites[0], device_type=devicetype, role=role),
+            Device(name='Device 1', site=sites[1], device_type=devicetype, role=role),
```
Are we intentionally reusing the `Device 1` name here? Seems like it's redundant and could lead to confusion later, but maybe I'm missing the intent.

If the intent is not purposefully re-using the name, then I think we should stick with test conventions and keep incrementing the index in the name.

pheus replied:
Yes, the reuse of the device name was intentional, as the issue specifically called for supporting devices with the same name across different sites. I thought including this scenario in the tests would help demonstrate and validate that requirement.

That said, I completely understand how this could cause confusion, and I’m happy to revert to the standard naming convention if preferred. :)

Thanks so much for taking the time to review the changes!

jnovinger commented on netbox/dcim/tests/test_views.py:
```diff
@@ -3328,12 +3332,12 @@ def setUpTestData(cls):
+            "Site 1,Device 3,dcim.interface,Interface 2,Site 2,Device 1,dcim.interface,Interface 2",
+            "Site 1,Device 3,dcim.interface,Interface 3,Site 2,Device 1,dcim.interface,Interface 3",
+            "Site 1,Device 1,dcim.interface,Device 2 Interface,Site 2,Device 1,dcim.interface,Interface 4",
+            "Site 1,Device 1,dcim.interface,Device 3 Interface,Site 2,Device 1,dcim.interface,Interface 5",
```
I think it's out of scope for this issue, but I think this is the sort of thing where we're starting to feel the lack of ability to test more than a single CSV dataset/shape.

I'm making a note to raise this as an FR.

pheus replied:
Absolutely. Thanks for being so supportive!

jnovinger reviewed (commented):
@pheus , overall, this looks great and acts as expected. I did have one question about changes to the test, if you could take a look.

Thanks!

pheus commented:
@jnovinger Thank you very much. I responded on your comments. Let me know, how you like to proceed 🙂

pheus committed:
docs(dcim): Update comments explaining indent for CSV import

Improved the inline comments to clarify the rationale behind allowing
devices with duplicate names on different sites during CSV bulk import.
